### ISSUE TITLE
JZColumnHeaderBackground and JZRowHeaderBackground is always white, and sticks out in sark mode.

### ISSUE THREAD

**Issue by gerkov77** [zjfjack/JZCalendarWeekView#121]
Please, I have no information about where I can access and set the background properties of these reusable views. Also, what method should be overwritten for changing these, or dequeuing my own subclasses for these items . 
If anyone could help me out here! 
Thanks!

**Comment by twilightDD** [comment]
@gerkov77 Thank you! 

I had the same problem and solved it like this. (It's probably the same way!)


```swift
class CalendarWeekView: JZBaseWeekView {
    
    //MARK: - JZBaseWeekView Methods    
    override func setup() {
        super.setup()
        collectionView.backgroundColor = UIColor.systemBackground // to support Dark Mode 
    }
    
    override func registerViewClasses() {
        super.registerViewClasses()
        
        // overwrite supplementary registers from super.registerViewClasses()
        collectionView.register(SOXColumnHeader.self,
                                forSupplementaryViewOfKind: JZSupplementaryViewKinds.columnHeader,
                                withReuseIdentifier: JZSupplementaryViewKinds.columnHeader)
        /* add register() for other supplementaryViews, too. See below.*/
        
        flowLayout.register(SOXColumnHeaderBackground.self,
                            forDecorationViewOfKind:  JZDecorationViewKinds.columnHeaderBackground)
        /* add register() for other decorationViews, too. See below.*/
    }
    
}

// Subclasses for Headers and Backgrounds
class SOXColumnHeader: JZColumnHeader {
    public override init(frame: CGRect) {
        super.init(frame: .zero)
        backgroundColor = .clear // the corresponding background-class will get the correct color
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}


class SOXColumnHeaderBackground: JZColumnHeaderBackground {
    public override init(frame: CGRect) {
        super.init(frame: frame)
        backgroundColor = UIColor.systemBackground // uses correct color on Dark Mode
    }
    required public init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}

```

I subclassed 
* JZColumnHeader
* JZCornerHeader
* JZRowHeader
* JZAllDayHeader

and the corresponding background classes 
* JZColumnHeaderBackground
* JZRowHeaderBackground
* JZAllDayHeaderBackground
* JZAllDayCorner

the same way as shown in the example above.

Best regards!
Peter


<img width="400" alt="Bildschirmfoto 2021-03-31 um 23 45 30" src="https://user-images.githubusercontent.com/792773/113215539-4919ea80-927b-11eb-8483-9d7de661dbec.png">

**Comment by spiccinocchi** [comment]
Great implementation, 
but there is a problem if you change the apparence from ligth to dark while you are using the app. Have you found a solution for this issue?

**Comment by twilightDD** [comment]
> Great implementation,
> but there is a problem if you change the apparence from ligth to dark while you are using the app. Have you found a solution for this issue?

Well, I don't exactly understand what you mean ...

**Comment by spiccinocchi** [comment]
If you start the calendar app while the device is in light mode and set your custom background color and then change it in dark mode, there is no way to change the background again. 
I hope is now more clear my problem

**Comment by twilightDD** [comment]
Thank you! I just works fine for me. 

(I _don't_ catch a notification to change the appearance, though.)

https://user-images.githubusercontent.com/792773/115515680-abb44400-a285-11eb-8630-a344a7ba9e7d.MP4

**Comment by spiccinocchi** [comment]
Yes you are right, it works for you beacuse you choose the systembackground color and it's correctly changed in dark appearance by iOS. 
The problem, for me, is that I don't catch a notification to change the appearance inside the class, so I can't use custum color for the background and also for the label.

**Comment by twilightDD** [comment]
> Yes you are right, it works for you beacuse you choose the systembackground color and it's correctly changed in dark appearance by iOS.

Which color do you use and in which subclass? A different system provided one or a complete custom?

**Comment by gerkov77** [comment]
> Yes you are right, it works for you because you choose the systembackground color and it's correctly changed in dark appearance by iOS.
> The problem, for me, is that I don't catch a notification to change the appearance inside the class, so I can't use custom colour for the background and also for the label.

I was using custom colours -no systemBackrgound stuff- setting two different colour set for dark and light mode. Just set your custom colours in assets and assign it to light and dark modes, then use them with the calendar - there shouldn't be any problem with that..
